src/csharp/Grpc.Tools/Common.cs

```diff
@@ -0,0 +1,105 @@
+#region Copyright notice and license
+
+// Copyright 2018 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security;
+
+[assembly: InternalsVisibleTo("Grpc.Tools.Tests")]
+
+namespace Grpc.Tools {
+  // Metadata names that we refer to often.
+  static class Metadata {
+    // On output dependency lists.
+    public static string kSource = "Source";
+    // On ProtoBuf items.
+    public static string kProtoRoot = "ProtoRoot";
+    public static string kOutputDir = "OutputDir";
+    public static string kGrpcServices = "GrpcServices";
+    public static string kGrpcOutputDir = "GrpcOutputDir";
+  };
+
+  // A few flags used to control the behavior under various platforms.
+  internal static class Platform {
+    public enum OsKind { Unknown, Windows, Linux, MacOsX };
+    public static readonly OsKind Os;
+
+    public enum CpuKind { Unknown, X86, X64 };
+    public static readonly CpuKind Cpu;
+
+    // This is not necessarily true, but good enough. BCL lacks a per-FS
+    // API to determine file case sensitivity.
+    public static bool IsFsCaseInsensitive => Os == OsKind.Windows;
+    public static bool IsWindows => Os == OsKind.Windows;
+
+    static Platform() {
```

Comment:
Dependencies in MSBuild assemblies just do not work. Microsoft's own implementation includes a set of common files into each assembly.

>  perhaps we could just include the source file in Grpc.Tools

Not without refactoring it significantly, because
```
#if NETSTANDARD1_5
```
and the MSBuild library is using the 1.3 API,
```
 [DllImport("libc")]
 static extern int uname(IntPtr buf);
```
Maybe safe from inside MSBuild, but I'd better not open this can of worms in this iteration. The Tools library needs only a very small subset of the PlatformApis.cs (strictly speaking, the only distinction in this library is made is between Windows and everything else).